tuf/pyca_crypto_keys.py

```diff
@@ -0,0 +1,1017 @@
+                        format=serialization.PrivateFormat.TraditionalOpenSSL,
+                        encryption_algorithm=serialization.NoEncryption())
+ 
+  # Need to generate the public pem from the private key before serialization
+  # to PEM.
+  public_key = private_key.public_key()
+  public_pem = public_key.public_bytes(encoding=serialization.Encoding.PEM,
+                    format=serialization.PublicFormat.SubjectPublicKeyInfo)
+ 
+  return public_pem.decode(), private_pem.decode()
+
+
+
+
+
+def create_rsa_signature(private_key, data):
+  """
+  <Purpose>
+    Generate an RSASSA-PSS signature.  The signature, and the method (signature
+    algorithm) used, is returned as a (signature, method) tuple.
+
+    The signing process will use 'private_key' to generate the signature of
+    'data'.
+
+    RFC3447 - RSASSA-PSS 
+    http://www.ietf.org/rfc/rfc3447.txt
+    
+    >>> public, private = generate_rsa_public_and_private(2048)
+    >>> data = 'The quick brown fox jumps over the lazy dog'.encode('utf-8')
+    >>> signature, method = create_rsa_signature(private, data)
+    >>> tuf.formats.NAME_SCHEMA.matches(method)
+    True
+    >>> method == 'RSASSA-PSS'
+    True
+    >>> tuf.formats.PYCRYPTOSIGNATURE_SCHEMA.matches(signature)
+    True
+
+  <Arguments>
+    private_key: 
+      The private RSA key, a string in PEM format.
+
+    data:
+      Data (string) used by create_rsa_signature() to generate the signature.
+
+  <Exceptions>
+    tuf.FormatError, if 'private_key' is improperly formatted.
+    
+    TypeError, if 'private_key' is unset.
+
+    tuf.CryptoError, if the signature cannot be generated. 
+
+  <Side Effects>
+    pyca/cryptography's 'RSAPrivateKey.signer()' called to generate the
+    signature.
+
+  <Returns>
+    A (signature, method) tuple, where the signature is a string and the method
+    is 'RSASSA-PSS'.
+  """
+  
+  # Does the arguments have the correct format?
+  # This check will ensure the arguments conform to 'tuf.formats.PEMRSA_SCHEMA'.
+  # and 'tuf.formats.DATA_SCHEMA' 
+  # Raise 'tuf.FormatError' if the checks fail.
+  tuf.formats.PEMRSA_SCHEMA.check_match(private_key)
+  tuf.formats.DATA_SCHEMA.check_match(data) 
+
+  # Signing 'data' requires a private key.  The 'RSASSA-PSS' signing method is
+  # the only method currently supported.
+  method = 'RSASSA-PSS'
+  signature = None
+
+  # Verify the signature, but only if the private key has been set.  The private
+  # key is a NULL string if unset.  Although it may be clearer to explicitly
+  # check that 'private_key' is not '', we can/should check for a value and not
+  # compare identities with the 'is' keyword.  Up to this point 'private_key'
+  # has variable size and can be an empty string.
+  if len(private_key):
+    
+    # pyca/cryptography's expected exceptions when generating RSA key object:
+    # "ValueError/IndexError/TypeError:  When the given key cannot be parsed
```

Comment:
Taking this at face value, doesn't the `try/except` block then let `IndexError`s raise? If so, this should be added to the docstring.